gradle/generate-pom.gradle

```diff
@@ -261,29 +264,143 @@ class ProjectDependenciesAsXml {
     private static boolean isExternal(Dependency dependency) {
         return AbstractExternalModuleDependency.isAssignableFrom(dependency.class)
     }
+}
+
+/**
+ * A project dependency with its scope.
+ *
+ * @see
+ * <a href="https://maven.apache.org/guides/introduction/introduction-to-dependency-mechanism.html#Dependency_Scope">
+ *     More on dependency scopes </a>.
+ *
+ */
+class DependencyWithScope implements Comparable<DependencyWithScope> {
+
+    private static Map<String, DependencyScope> KNOWN_SCOPES
+
+    private final Dependency dependency
+    private final DependencyScope scope
+    private final Configuration tempConfig
+
+    static {
```

Comment:
Please document. Obviously, there are "scopes" and "scopes", and they differ in some way.